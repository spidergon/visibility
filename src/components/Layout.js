import React from 'react'
// import { setConfig } from 'react-hot-loader'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import { StaticQuery, graphql } from 'gatsby'
import { ThemeProvider } from 'styled-components'
import { theme, Style } from './styles/Style'
import { GlobalStateProvider } from '../lib/state'
import Header from './Header'
import Snack from './Snack'
import '../lib/cookieconsent'
// setConfig({ pureSFC: true })

const Layout = ({ children }) => (
  <ThemeProvider theme={theme}>
    <StaticQuery
      query={graphql`
        query SiteTitleQuery {
          site {
            siteMetadata {
              title
            }
          }
        }
      `}
      render={data => (
        <GlobalStateProvider>
          <Helmet>
            <link
              href='//cdnjs.cloudflare.com/ajax/libs/cookieconsent2/3.1.0/cookieconsent.min.css'
              rel='stylesheet'
              type='text/css'
            />
            <link
              crossOrigin='anonymous'
              href='https://use.fontawesome.com/releases/v5.6.1/css/all.css'
              integrity='sha384-gfdkjb5BdAXd+lj+gudLWI+BXq4IuLW5IT+brZEZsLFm++aCMlF1V92rMkPaX4PP'
              rel='stylesheet'
            />
            <link
              href='https://fonts.googleapis.com/css?family=Roboto'
              rel='stylesheet'
            />
          </Helmet>
          <Style />
          <Snack />
          <Header siteTitle={data.site.siteMetadata.title} />
          <main>{children}</main>
        </GlobalStateProvider>
      )}
    />
  </ThemeProvider>
)

Layout.propTypes = {
  children: PropTypes.node.isRequired
}

export default Layout
