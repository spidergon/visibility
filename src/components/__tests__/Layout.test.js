import React from 'react'
import renderer from 'react-test-renderer'
import { StaticQuery } from 'gatsby' // mocked

import Layout from '../Layout'

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          title: `Visibility`
        }
      }
    })
  )
})

describe('<Layout/>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Layout>
          <h1>{'Hey!'}</h1>
        </Layout>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
