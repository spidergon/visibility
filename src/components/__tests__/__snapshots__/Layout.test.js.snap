// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout/> renders correctly 1`] = `
Array [
  <div
    className="sc-bxivhb fzeEHn"
  />,
  <header
    className="sc-bdVaJa iUtKEW"
  >
    <div
      className="content grid"
    >
      <nav
        className="logo"
      >
        <a
          href="/"
        >
          <span>
            Visibility
          </span>
        </a>
      </nav>
      <nav
        className="navs"
      />
      <nav
        className="profile"
      >
        <ul>
          <li>
            <a
              className="link"
              href="/connexion"
            >
              <div
                className="sub"
              >
                <span>
                  Connexion
                </span>
              </div>
            </a>
          </li>
        </ul>
        <div
          className="offline"
        >
          <span
            className="menu-anchor"
            onClick={[Function]}
          >
            <i
              className="fas fa-bars"
            />
          </span>
        </div>
      </nav>
    </div>
  </header>,
  <main>
    <h1>
      Hey!
    </h1>
  </main>,
]
`;
